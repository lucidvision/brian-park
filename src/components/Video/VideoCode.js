import React from 'react'
import styled from 'styled-components'
import videomp4 from 'assets/HelloWorld.mp4'
import videowebm from 'assets/HelloWorld.webm'

const StyledVideo = styled.video`
  height: 100%;
  width: 100%;
  object-fit: cover;
  z-index: -1;
`

export default function VideoCode (props) {
  return (
    <StyledVideo autoPlay muted loop {...props}>
      <source src={videomp4} type='video/mp4' />
      <source src={videowebm} type='video/webm' />
      Your browser is not supported!
    </StyledVideo>
  )
}
