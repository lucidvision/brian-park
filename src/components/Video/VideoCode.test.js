import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import VideoCode from './VideoCode'
import videomp4 from 'assets/HelloWorld.mp4'
import videowebm from 'assets/HelloWorld.webm'

Enzyme.configure({adapter: new Adapter()})

describe('VideoCode', () => {
  const videoCode = shallow(<VideoCode />)

  it('renders correctly', () => {
    expect(videoCode).toMatchSnapshot()
  })

  describe('renders source with the correct props', () => {
    it('renders two sources', () => {
      expect(videoCode.find('source')).toHaveLength(2)
    })

    it('renders a mp4 source', () => {
      expect(videoCode.find('source').at(0).props()).toEqual({
        src: videomp4,
        type: 'video/mp4'
      })
    })

    it('renders a webm source', () => {
      expect(videoCode.find('source').at(1).props()).toEqual({
        src: videowebm,
        type: 'video/webm'
      })
    })
  })
})
