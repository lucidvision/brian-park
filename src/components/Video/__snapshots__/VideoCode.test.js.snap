// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoCode />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoPlay": true,
      "children": Array [
        <source
          src="HelloWorld.mp4"
          type="video/mp4"
/>,
        <source
          src="HelloWorld.webm"
          type="video/webm"
/>,
        "Your browser is not supported!",
      ],
      "loop": true,
      "muted": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "HelloWorld.mp4",
          "type": "video/mp4",
        },
        "ref": null,
        "rendered": null,
        "type": "source",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "HelloWorld.webm",
          "type": "video/webm",
        },
        "ref": null,
        "rendered": null,
        "type": "source",
      },
      "Your browser is not supported!",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoPlay": true,
        "children": Array [
          <source
            src="HelloWorld.mp4"
            type="video/mp4"
/>,
          <source
            src="HelloWorld.webm"
            type="video/webm"
/>,
          "Your browser is not supported!",
        ],
        "loop": true,
        "muted": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "HelloWorld.mp4",
            "type": "video/mp4",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "HelloWorld.webm",
            "type": "video/webm",
          },
          "ref": null,
          "rendered": null,
          "type": "source",
        },
        "Your browser is not supported!",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
